ibtk/src/lagrangian/FEProjector.cpp

```diff
@@ -329,7 +332,10 @@ FEProjector::buildDiagonalL2MassMatrix(const std::string& system_name)
                 // for spread forces and other places where we already have a
                 // finite element solution vector and are not solving a linear
                 // system.
-                M_vec->add_vector(M_e_vec, dof_indices[var_num]);
+                dof_id_scratch.clear();
```

Comment:
It is too bad that we have to have these verbose set of statements to reset the `dof_id_scratch` vectors. If `dof_id_scratch` already is the right size, could we do a `copy` instead?

Reply:
Unfortunately the constraint code will resize the vector, so we don't have any guarantees that the scratch vector will have the right size at this point.

Reply:
It seems common enough that it might be worth having a helper function to turn this in to a self-documenting one-liner?

Reply:
Let me see if that ends up being shorter - I'll give it a try.

Reply:
That is better.